modules/public/views.py

```diff
@@ -202,14 +202,16 @@ def snap_details(snap_name):
 
     # filter out banner and banner-icon images from screenshots
     screenshots = [
-        url for url in details['screenshot_urls'] if "banner" not in url
+        m['url'] for m in details['media']
+        if m['type'] == "screenshot" and "banner" not in m['url']
```

Comment:
@cprov, @bartaz is `"banner" not in m['url']` still relevant? Is `m['type'] == "screenshot"` not enough to filter out banners?

Reply:
@nottrobin the filtering on file name still needed until the existing (~5) snaps migrate their banner* and banner-icon* out of the `screenshot` type. But they will only do that once they have UI available.